components/board/collectors_test.go

```diff
@@ -82,7 +83,13 @@ func TestCollectors(t *testing.T) {
 			defer col.Close()
 			col.Collect()
 			mockClock.Add(captureInterval)
-			test.That(t, buf.Length(), test.ShouldEqual, 1)
+
+			length := 0
+			for i := 0; i < numRetries && length == 0; i++ {
+				length = buf.Length()
+				time.Sleep(time.Second)
+			}
+			test.That(t, length, test.ShouldBeGreaterThan, 0)
```

Comment:
It would be great if we could pull this out into a testutil function of our own! Could take a condition to check, plus a numRetries value, and sleep numRetries time until the condition is true